src/IO/WriteBufferFromVector.h

```diff
@@ -83,6 +85,13 @@ class WriteBufferFromVector : public WriteBuffer
         size_t old_size = vector.size();
         /// pos may not be equal to vector.data() + old_size, because WriteBuffer::next() can be used to flush data
         size_t pos_offset = pos - reinterpret_cast<Position>(vector.data());
+        size_t new_size = old_size * size_multiplier;
+        if (max_memory_usage != 0 && new_size > max_memory_usage)
```

Comment:
By the way, checking `new_size` against `max_memory_usage` is not correct, since memory can be used for anything, so at least it is better to use `CurrentThread::getMemoryTracker()->get()` here.

But I would better to remove this, since tracking should not be done in this way.